### PR TITLE
Add full screen support for Android 14

### PR DESCRIPTION
**Pull request by plavelo**

### CODE REVIEW

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt [hunk 1]
```diff
@@ -186,7 +186,11 @@ class CallkitNotificationManager(private val context: Context) {
                     .into(targetLoadAvatarDefault)
             }
             val caller = data.getString(CallkitConstants.EXTRA_CALLKIT_NAME_CALLER, "")
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            val notificationManager: NotificationManager =
+                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            val canUseFullScreenIntent =
+                Build.VERSION.SDK_INT < Build.VERSION_CODES.UPSIDE_DOWN_CAKE || notificationManager.canUseFullScreenIntent()
```
Comment (AAkira) [inline android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt]: This is incorrect. 

`Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE && notificationManager.canUseFullScreenIntent()`

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt [hunk 2]
```diff
@@ -186,7 +186,11 @@ class CallkitNotificationManager(private val context: Context) {
...
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && canUseFullScreenIntent) {
```
Comment (AAkira) [inline android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt]: This block seems to set the only CallStyle.
It is not related to full screen.

You should check this here(L125).

https://github.com/AAkira/flutter_callkit_incoming/pull/24/files#diff-0ee07ecfd0bb91b3a7aaf8f1c530fbc1cebaf45b78539854cfd2256446c24d49R125

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt [hunk 3]
```diff
@@ -280,6 +284,39 @@ class FlutterCallkitIncomingPlugin : FlutterPlugin, MethodCallHandler, ActivityA
                     }
                     callkitNotificationManager?.requestNotificationPermission(activity, map)
                 }
+
+                "canUseFullScreenIntent" -> {
+                    val notificationManager: NotificationManager =
+                        context?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+                    val canUseFullScreenIntent =
+                        Build.VERSION.SDK_INT < Build.VERSION_CODES.UPSIDE_DOWN_CAKE || notificationManager.canUseFullScreenIntent()
```
Comment (AAkira) [inline android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt:294]: This is incorrect too.

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt [hunk 4]
```diff
@@ -280,6 +284,39 @@ class FlutterCallkitIncomingPlugin : FlutterPlugin, MethodCallHandler, ActivityA
...
+                    result.success(canUseFullScreenIntent)
+                }
+
+                "openFullScreenIntentSettings" -> {
+                    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                        return
+                    }
+                    val packageName = call.arguments as? String ?: ""
```
Comment (AAkira) [inline android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt]: It is preferable to get a package name from `context.packageName`.

MethodCall.arguments is not for getting a package name.
https://api.flutter.dev/javadoc/io/flutter/plugin/common/MethodCall.html#arguments

And I would prefer an early return if a context is null.

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt (hunk 1 above)
Comment (plavelo) [reply android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt]: According to the documentation, apps always have the USE_FULL_SCREEN_INTENT permission on Android 14 or earlier. So canUseFullScreenIntent should be true on Android 14 or earlier.

`Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE && notificationManager.canUseFullScreenIntent()`
This code returns false on Android 14 or earlier.

https://developer.android.com/reference/android/app/NotificationManager#canUseFullScreenIntent()
```
From Android Build.VERSION_CODES.UPSIDE_DOWN_CAKE, apps may not have permission to use Manifest.permission.USE_FULL_SCREEN_INTENT
```

Comment (AAkira) [reply android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt]: I'm sorry.
I took this to mean whether this full-screen feature is enabled or not.

It's ok.

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt (hunk 2 above)
Comment (plavelo) [reply android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/CallkitNotificationManager.kt]: I have reverted this change.

When removing the line `<uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT"/>` from AndroidManifest.xml, the following error occurred.

```
Not posted. CallStyle notifications must be for a foreground service or user initated job or use a fullScreenIntent.
```

However, I have reverted the USE_FULL_SCREEN_INTENT line in AndroidManifest.xml so this fix is no longer necessary.

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt (hunk 4 above)
Comment (plavelo) [reply android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt]: I fixed it 🙏

File: android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt [hunk 5]
```diff
@@ -280,6 +284,40 @@ class FlutterCallkitIncomingPlugin : FlutterPlugin, MethodCallHandler, ActivityA
...
+                    try {
+                        ctx.startActivity(
+                            Intent(
+                                Settings.ACTION_MANAGE_APP_USE_FULL_SCREEN_INTENT,
+                                Uri.parse("package:${ctx.packageName}"),
+                            )
+                                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_NO_HISTORY)
+                        )
+                    } catch (e: ActivityNotFoundException) {
+                        context?.startActivity(
```
Comment (AAkira) [inline android/src/main/kotlin/com/hiennv/flutter_callkit_incoming/FlutterCallkitIncomingPlugin.kt]:
```suggestion
                        ctx.startActivity(
```

**Review by AAkira — APPROVED** [review]

**Pull request merged**